hudi-utilities/src/main/java/org/apache/hudi/utilities/sources/helpers/IncrSourceHelper.java

```diff
@@ -183,7 +183,8 @@ public static Pair<CloudObjectIncrCheckpoint, Option<Dataset<Row>>> filterAndGen
                                                                                                                     long sourceLimit, QueryInfo queryInfo,
                                                                                                                     CloudObjectIncrCheckpoint cloudObjectIncrCheckpoint) {
     if (sourceData.isEmpty()) {
-      return Pair.of(cloudObjectIncrCheckpoint, Option.empty());
+      // There is no file matching the prefix.
+      return Pair.of(new CloudObjectIncrCheckpoint(queryInfo.getEndInstant(), null), Option.empty());
```

Comment:
Why can't we use the cloudObjectIncrCheckpoint that is being passed to this function ?

Reply:
cloudObjectIncrCheckpoint should not be used here to allow progress of checkpoints where there are empty commits.